src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js

```diff
@@ -12,20 +12,28 @@ const { APP, JOB_ID } = CHAOS_EXPERIMENT_LABELS;
 const STATUS_TO_CLEAN = 'Stop';
 let supportedChaosKinds;
 
-if (kubernetesConfig.kubernetesToken) {
-    logger.info('Using kubernetes token from env var');
-    headers.Authorization = 'bearer ' + kubernetesConfig.kubernetesToken;
-} else {
+const init = async () => {
+    if (kubernetesConfig.kubernetesToken) {
+        logger.info('Using kubernetes token from env var');
+        headers.Authorization = 'bearer ' + kubernetesConfig.kubernetesToken;
+    } else {
+        try {
+            logger.info('kubernetes token from env var was not provided. will use: ' + TOKEN_PATH);
+            const token = fs.readFileSync(TOKEN_PATH);
+            headers.Authorization = 'bearer ' + token.toString();
+        } catch (error) {
+            logger.warn(error, 'Failed to get kubernetes token from: ' + TOKEN_PATH);
+        }
+    }
     try {
-        logger.info('kubernetes token from env var was not provided. will use: ' + TOKEN_PATH);
-        const token = fs.readFileSync(TOKEN_PATH);
-        headers.Authorization = 'bearer ' + token.toString();
-    } catch (error) {
-        logger.warn(error, 'Failed to get kubernetes token from: ' + TOKEN_PATH);
+        supportedChaosKinds = await getSupportedKinds();
+    } catch (e){
+        logger.error('Failed to get kubernetes custom defined resources, make sure privileges are configured correctly', e);
```

Comment:
error might not relate only to privileges. it might be because chaos-mesh is not installed

Reply:
Not installed but they came up with the boolean to use chaos? Not sure if I want to log that, you think needed?